src/Repo_new.java

```diff
@@ -0,0 +1,804 @@
+    private volatile List<Pattern> _ignoreFiles;
+    private View _view;
+    private Folder _rootFolder;
+    private File _tempDir;
+
+    public Repo(final RepoProperties repoProperties, final Logger logger) {
+        _branchName = repoProperties.getBranchName();
+        _userNamePattern = repoProperties.getProperty(PROP_USER_PATTERN,
+                PROP_DEFAULT_USER_PATTERN);
+        _repoProperties = repoProperties;
+        _logger = logger;
+        _pool = new ConnectionPool();
+        _viewCache = new ArrayList<>();
+        _folderCache = new ConcurrentHashMap<>();
+        _fileCache = new ConcurrentHashMap<>();
+        //@formatter:off
+        _commentFormat = new MessageFormat(
+           "{0}\n\n"
+         + FOOTER_START + '\n'
+         + "{1}\n"
+         + FOOTER_END);
+      //@formatter:on
+    }
+
+    public Server getIdleConnection() {
+        return _pool.get();
+    }
+
+    public void releaseConnection(final Server s) {
+        _pool.release(s);
+    }
+
+    private Server createNewConnection() {
+        final StarTeamURL url = getUrl();
+        final RepoProperties props = getRepoProperties();
+        final int port = Integer.parseInt(url.getPort());
+        final int protocol = url.getProtocol();
+        final String host = url.getHostName();
+        final String ca = props.getProperty(PROP_CA, null);
+        final ServerInfo info = new ServerInfo();
+        String userName = props.getProperty(PROP_USER, null);
+        String password = props.getProperty(PROP_PASSWORD, null);
+        final Server server;
+
+        if (userName == null) {
+            userName = url.getUserName();
+        }
+        if (password == null) {
+            password = url.getPassword();
+        }
+        if (ca != null) {
+            if (Boolean.parseBoolean(ca)) {
+                info.setAutoLocateCacheAgent(true);
+            } else {
+                final int ind = ca.indexOf(':');
+
+                if (ind == -1) {
+                    info.setMPXCacheAgentAddress(ca);
+                } else {
+                    info.setMPXCacheAgentAddress(ca.substring(0, ind));
+                    info.setMPXCacheAgentPort(Integer.parseInt(ca
+                            .substring(ind + 1)));
+                }
+            }
+        }
+
+        info.setHost(host);
+        info.setPort(port);
+        info.setCompression(true);
+        info.setConnectionType(protocol);
+        info.setMPXCacheAgentThreadCount(Integer.parseInt(props.getProperty(
+                PROP_CATHREADS, PROP_DEFAULT_CATHREADS)));
+        info.setEnableCacheAgentForFileContent(ca != null);
+        info.setEnableCacheAgentForObjectProperties(ca != null);
+        server = new Server(info);
+        server.connect();
+
+        if ((userName != null) && (password != null)) {
+            if (server.logOn(userName, password) == 0) {
+                throw new ConfigurationException("Failed to login to " + host
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Failed to login to " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1192)